metrics/http.go

```diff
@@ -20,9 +23,58 @@ type (
 	}
 )
 
+// HTTPEndpointDetails provides information about the endpoint, using each attribute as a label in the metric.
+type HTTPEndpointDetails struct {
+	// Path represents the relative path pattern (i.e. /api/v1/status)
+	Path string
+	// Verb represents the verb used on the endpoint resource (i.e. GET)
+	Verb string
+}
+
+// InitMetricDetails includes details about the server host, status codes, and path/verb for each endpoint.
+// This is used to figure out which specific metric combination to initialize.
+type InitMetricDetails struct {
+	// EndpointDetails are the HTTP details for each Path + Verb combo.
+	EndpointDetails []*HTTPEndpointDetails
+	// Host is the host of the actual running server.
+	Host string
+	// StatusCodes are the set of status codes that are possible for the endpoints (i.e. 400, 500, 200)
```

Comment:
That breakdown does not seem quite right: different endpoints may return different status codes.

Reply:
Yes, you're correct about this. I think what I would have done to change this is make each endpoint detail have its own relevant status codes. This was originally meant to solve the problem of metrics not being initialized for each label combination (status + HTTP path)